features/admin.application-templates.v1/components/application-tab-components-filter.tsx

```diff
@@ -0,0 +1,104 @@
+import { Grid } from "semantic-ui-react";
+import { ApplicationTabIDs } from "../../admin.extensions.v1";
+import useApplicationTemplateMetadata from "../hooks/use-application-template-metadata";
+import { ApplicationEditTabMetadataInterface } from "../models/templates";
+
+/**
+ * Prop types of the `ApplicationTabComponentsFilter` component.
+ */
+export interface ApplicationTabComponentsFilterPropsInterface extends IdentifiableComponentInterface {
+    /**
+     * Current tab id.
+     */
+    tabId: ApplicationTabIDs;
+}
+
+/**
+ * Application tab components filtering component.
+ *
+ * @param Props - Props to be injected into the component.
+ */
+export const ApplicationTabComponentsFilter: FunctionComponent<
+    PropsWithChildren<ApplicationTabComponentsFilterPropsInterface>
+> = ({
+    tabId,
+    children,
+    "data-componentid": _componentId = "application-tab-components-filter"
+}: PropsWithChildren<ApplicationTabComponentsFilterPropsInterface>): ReactElement => {
+
+    const {
+        templateMetadata,
+        isTemplateMetadataRequestLoading
+    } = useApplicationTemplateMetadata();
+
+    /**
+     * Extract the data-component IDs that need to be hidden in the current application tab.
+     */
+    const hiddenComponents: string[] = useMemo(() => {
+        let componentList: string[] = [];
+
+        templateMetadata?.edit?.tabs?.forEach((tab: ApplicationEditTabMetadataInterface) => {
+            if (tab?.id === tabId) {
+                if (tab?.hiddenComponents
+                    && Array.isArray(tab?.hiddenComponents)
+                    && tab?.hiddenComponents?.length > 0) {
+                    componentList = tab?.hiddenComponents;
+                }
+            }
+        });
+
+        return componentList;
+    }, [ templateMetadata, tabId ]);
+
+    return (
+        isTemplateMetadataRequestLoading || !hiddenComponents
+            ? (
+                <Grid.Row columns={ 1 }>
+                    <Grid.Column mobile={ 16 } tablet={ 16 } computer={ 16 }>
+                        <ContentLoader inline="centered" active/>
+                    </Grid.Column>
+                </Grid.Row>
+            )
+            : hiddenComponents?.length === 0
+                ? (
+                    <>
+                        { children }
+                    </>
+                ) : (
+                    <>
+                        {
+                            React.Children.map(children, (child: ReactNode) => {
+                                const componentId: string = child?.["props"]?.["data-componentid"];
+
+                                if (componentId && hiddenComponents?.includes(componentId)) {
+                                    return null;
+                                }
+
+                                return child;
+                            })
+                        }
+                    </>
+                )
+    );
```

Comment:
Shall we move this to if-else blocks for better readability?

Reply:
https://github.com/wso2/identity-apps/pull/5896/commits/33326430915c8312dc8113db6bcb65860c16c9ff